opensearch/src/main/java/org/opensearch/sql/opensearch/executor/OpenSearchExecutionEngine.java

```diff
@@ -108,6 +114,54 @@ public ExplainResponseNode visitTableScan(
         });
   }
 
+  private Hook.Closeable getPhysicalPlanInHook(AtomicReference<String> physical) {
+    return Hook.PLAN_BEFORE_IMPLEMENTATION.addThread(
```

Comment:
Calcite Javadoc says this Hook API is for testing and debugging?

Reply:
This seems the only way we could extract the physical plan and generated java code of linq4j expression in Calcite because we build our AST, rather than the Calcite SqlNode. And the HOOK APIs are stable and existing from 2013. IMO, it's okey to leverage this interface.